classic_tetris_project/util/cache.py

```diff
@@ -21,3 +22,6 @@ def has(self, filename):
     def put(self, filename, content):
         with open(self.full_path(filename), "wb") as f:
             f.write(content)
+    
+    def get(self, filename):
+        return open(self.full_path(filename), "rb")
```

Comment:
This file needs to be closed after it's opened. For now we want to read the entire file into memory, so it's easiest to use `open` as a context manager which closes the file descriptor at the end of the block:
```python
with open(self.full_path(filename), "rb") as f:
    return f.read()
```

Reply:
That's what I originally had but the discordpy API doesn't like it since it wants a file object.  I moved to a iobytes object (read into iobytes then close file and return iobytes) and then back to this.  Should I revert back to iobytes object, or should I return f.read then caller needs to wrap in iobytes object?

Reply:
It doesn't look like `HzSimulation#get_image` is used anywhere. Is this just something you're using locally for your testing?

I think returning a `BytesIO` object is probably fine for now. The alternative would be to define a method like `FileCache#stream` used as a context manager to expose the `File` object and close it at the end of the `with` block.

Reply:
Get_image is used for testing since we can skip the cache if we call it instead.; If like to leave it in but I guess people could call the wrong function?